midend/parserUnroll.cpp

```diff
@@ -625,26 +625,19 @@ class ParserSymbolicInterpreter {
                 auto packets = state->before->filter(filter);
                 auto prevPackets = crt->before->filter(filter);
                 if (packets->equals(prevPackets)) {
-                    bool conservative = false;
                     for (auto p : state->before->map) {
                         if (p.second->is<SymbolicPacketIn>()) {
                             auto pkt = p.second->to<SymbolicPacketIn>();
                             if (pkt->isConservative()) {
-                                conservative = true;
                                 break;
                             }
                         }
                     }
-
-                    if (conservative)
-                        ::warning(ErrorType::WARN_PARSER_TRANSITION,
-                                    "Potential parser cycle without extracting any bytes:\n%1%",
-                                    stateChain(state));
-                    else
+                    if (equStackVariableMap(crt->statesIndexes, state->statesIndexes)) {
                         ::warning(ErrorType::ERR_INVALID,
-                                  "Parser cycle without extracting any bytes:\n%1%",
```

Comment:
this is ok, but is advance considered?

Reply:
The warning message will be printed for an "advance" without an explanation of the real reason:
```
[--Wwarn=invalid] warning: Parser cycle can't be unrolled, because ParserUnroll can't detect the number of loop iterations:
Parser ParserI state chain: start, s1, s2, s2
```
I don't think that catching an "advance" here is a good solution because it is considered in Interpreter class only. ParserUnroll uses this class for the calculation of the values of header stack indexes only.  The real reason why this loop can't be unrolled with ParserUnroll is that the states of the header stack indexes are the same for the last iteration of the loop. In general, it could be any reason for that, not just "advance". I can't build a complete list of the reasons for that. So, I decide to make the warning message more general.